packages/twenty-front/src/modules/workflow/components/RightDrawerWorkflowEditStepContent.tsx

```diff
@@ -0,0 +1,93 @@
+import { RightDrawerWorkflowEditStepContentAction } from '@/workflow/components/RightDrawerWorkflowEditStepContentAction';
+import { RightDrawerWorkflowEditStepContentTrigger } from '@/workflow/components/RightDrawerWorkflowEditStepContentTrigger';
+import { TRIGGER_STEP_ID } from '@/workflow/constants/TriggerStepId';
+import { useUpdateWorkflowVersionStep } from '@/workflow/hooks/useUpdateWorkflowVersionStep';
+import { useUpdateWorkflowVersionTrigger } from '@/workflow/hooks/useUpdateWorkflowVersionTrigger';
+import { showPageWorkflowSelectedNodeState } from '@/workflow/states/showPageWorkflowSelectedNodeState';
+import { WorkflowWithCurrentVersion } from '@/workflow/types/Workflow';
+import { findStepPositionOrThrow } from '@/workflow/utils/findStepPositionOrThrow';
+import { useRecoilValue } from 'recoil';
+import { isDefined } from 'twenty-ui';
+
+const getStepDefinition = ({
+  stepId,
+  workflow,
+}: {
+  stepId: string;
+  workflow: WorkflowWithCurrentVersion;
+}) => {
+  const currentVersion = workflow.currentVersion;
+  if (!isDefined(currentVersion)) {
+    return undefined;
+  }
+
+  if (stepId === TRIGGER_STEP_ID) {
+    if (!isDefined(currentVersion.trigger)) {
+      return undefined;
+    }
+
+    return {
+      type: 'trigger',
+      definition: currentVersion.trigger,
+    } as const;
+  }
+
+  if (!isDefined(currentVersion.steps)) {
+    return undefined;
+  }
+
+  const selectedNodePosition = findStepPositionOrThrow({
+    steps: currentVersion.steps,
+    stepId: stepId,
+  });
+
+  return {
+    type: 'action',
+    definition: selectedNodePosition.steps[selectedNodePosition.index],
+  } as const;
+};
+
+export const RightDrawerWorkflowEditStepContent = ({
+  workflow,
+}: {
+  workflow: WorkflowWithCurrentVersion;
+}) => {
+  const showPageWorkflowSelectedNode = useRecoilValue(
+    showPageWorkflowSelectedNodeState,
+  );
+  if (!isDefined(showPageWorkflowSelectedNode)) {
+    throw new Error(
+      'Expected a node to be selected. Selecting a node is mandatory to edit it.',
+    );
+  }
+
+  const { updateTrigger } = useUpdateWorkflowVersionTrigger({ workflow });
+  const { updateStep } = useUpdateWorkflowVersionStep({
+    workflow,
+    stepId: showPageWorkflowSelectedNode,
+  });
+
+  const stepConfiguration = getStepDefinition({
+    stepId: showPageWorkflowSelectedNode,
```

Comment:
same for all those states, I don't think we usually but the page in there. I see you have a lot of states named like this so we need to find better naming

Reply:
Done